src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/NuGet/NuGetRestorer.cs

```diff
@@ -98,7 +104,11 @@ protected override Task DisposeCoreAsync(bool initialized)
             // active configuration should be updated before resetting subscriptions
             await RefreshActiveConfigurationAsync().ConfigureAwait(false);
 
-            _designTimeBuildSubscriptionLink?.Dispose();
+            foreach (var disposable in _designTimeBuildSubscriptionLinks)
+            {
+                disposable.Dispose();
+            }
+            _designTimeBuildSubscriptionLinks.Clear();
```

Comment:
This looks like a race condition here. `ResetSubscriptions` and `Dispose` could enumerate and clear the list at the same time.
Also, `ResetSubscriptionsAsync`